source/patterns/@aws-solutions-constructs/core/test/secretsmanager-helper.test.ts

```diff
@@ -0,0 +1,64 @@
+/**
+ *  Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ *  with the License. A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
+ *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
+ *  and limitations under the License.
+ */
+
+import {RemovalPolicy, Stack} from '@aws-cdk/core';
+import * as defaults from '../';
+import {ResourcePart, SynthUtils} from '@aws-cdk/assert';
+import '@aws-cdk/assert/jest';
+
+const DESCRIPTION = 'test secret description';
+const SECRET_NAME = 'test secret name';
+
+// --------------------------------------------------------------
+// Test minimal deployment with no properties
+// --------------------------------------------------------------
+test('Test minimal deployment with no properties', () => {
+  // Stack
+  const stack = new Stack();
+  // Helper declaration
+  defaults.buildSecretsManagerSecret(stack, 'secret', {});
+  // Assertion 1
+  expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
+  // Assertion 2
+  expect(stack).toHaveResourceLike('AWS::SecretsManager::Secret', {
+    Type: 'AWS::SecretsManager::Secret',
+    UpdateReplacePolicy: 'Retain',
+    DeletionPolicy: 'Retain'
+  }, ResourcePart.CompleteDefinition);
+});
+
+// --------------------------------------------------------------
+// Test deployment w/ custom properties
+// --------------------------------------------------------------
+test('Test deployment with custom properties', () => {
+  // Stack
+  const stack = new Stack();
+  // Helper declaration
+  defaults.buildSecretsManagerSecret(stack, 'secret', {
+    secretName: SECRET_NAME,
+    description: DESCRIPTION,
+    removalPolicy: RemovalPolicy.DESTROY,
+  });
+  // Assertion 1
+  expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
```

Comment:
Only use snapshot for 1 test (probably all defaults) - for the rest just use the inspection functions to check the specific target of the test. (My first construct I used snapshots everywhere, Hitendra suggested not to do that because refreshing snapshots can be onerous - in the time since then I have learned exactly what he means. :-) )